components/SwapStats.tsx

```diff
@@ -0,0 +1,149 @@
+import { useEffect, VFC } from "react";
+import { IntrinsicElements } from "@/types";
+import { LinearProgress, Tooltip, Theme } from "@mui/material";
+import InfoOutlinedIcon from "@mui/icons-material/InfoOutlined";
+import { css } from "@emotion/react";
+import { formatBalance } from "@/utils";
+import { useSwap } from "@/providers/SwapProvider";
+import { useSwapExchangeRate, useSwapGasFee } from "@/hooks";
+
+interface SwapStatsProps {}
+
+const SwapStats: VFC<IntrinsicElements["div"] & SwapStatsProps> = ({
+	...props
+}) => {
+	const { exchangeValue, exchangeAsset, receiveAsset, slippage, txStatus } =
+		useSwap();
+
+	const [exchangeRate, updateExchangeRate] = useSwapExchangeRate();
+	const [gasFee, gasAsset, updateGasFee] = useSwapGasFee();
+
+	useEffect(() => {
+		if (txStatus?.status !== "success") return;
+		updateExchangeRate();
+		updateGasFee();
+	}, [txStatus?.status, updateExchangeRate, updateGasFee]);
+
+	return (
+		<div {...props} css={styles.root}>
+			<LinearProgress
+				css={[styles.formInfoProgress(!!exchangeRate && !!gasFee)]}
+			/>
+			<ul>
+				<li>
+					<strong>Exchange Rate:</strong>{" "}
+					{!!exchangeRate && (
+						<span>
+							1 {exchangeAsset.symbol} &asymp; {formatBalance(exchangeRate)}{" "}
+							{receiveAsset.symbol}
+						</span>
+					)}
+					{!exchangeRate && <span>-</span>}
+				</li>
+				<li>
+					<strong>Gas Fee:</strong>{" "}
+					{!!gasFee && (
+						<span>
+							&asymp;{gasFee} {gasAsset.symbol}
+						</span>
+					)}
+					{!gasFee && <span>-</span>}
+				</li>
+				<li>
+					<strong>Slippage:</strong>{" "}
+					<span>
+						{formatBalance(
+							Number(exchangeValue.value) * (1 + Number(slippage) / 100)
+						)}{" "}
+						{exchangeAsset.symbol}
+					</span>
+					<Tooltip
+						disableFocusListener
+						PopperProps={
+							{
+								sx: styles.formInfoTooltip,
+							} as any
+						}
+						title={
+							<div>
+								If the amount of {exchangeAsset.symbol} used for swapping is
+								greater than Slippage value, the transaction will not proceed.
+								You can update the Slippage percentage under Settings.
+							</div>
+						}
+						arrow
+						placement="right"
+					>
+						<InfoOutlinedIcon fontSize={"0.5em" as any} />
+					</Tooltip>
+				</li>
+			</ul>
+		</div>
+	);
+};
+
+export default SwapStats;
+
+const styles = {
+	root: ({ palette }: Theme) => css`
+		margin-top: 2em;
+		padding: 1.5em;
+		color: ${palette.text.primary};
+		background-color: ${palette.background.main};
+		position: relative;
+
```

Comment:
might be worth just chucking a border radius here to match styling of other components